Joedmin-hw3/Joedmin-hw3/Program.cs

```diff
@@ -0,0 +1,153 @@
+                        {
+                            if (!(string.Equals(credentials[1], usersAndPasswords[credentials[0]])))
+                                Console.WriteLine("Invalid password.");
+                            else
+                            {
+                                Console.WriteLine("Hello!");
+                            }
+                        }
+                        else
+                            Console.WriteLine("Username is not registered.");
+                    }
+                }
+
+                // Register
+                else if (String.Equals(answer, "register", StringComparison.OrdinalIgnoreCase))
+                {
+                    Dictionary<string, string> usersAndPasswords = GetCredentials("Users.txt");
+                    using (var stream = new StreamWriter("Users.txt", true))
+                    {
+                        string[] credentials = GetCredencials();
+                        SaveToDictionary(usersAndPasswords, credentials);
+                        stream.WriteLine($"{credentials[0]} {credentials[1]}");
+                        Console.WriteLine("Hello!");
+                    }
+                }
+            }
+            catch (UserNotFoundException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            catch (DuplicateUserCredentialsException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            catch (FileNotFoundException)
+            {
+                Console.WriteLine("File Users.txt was not found. Make sure it is in the same directory as the .exe file.");
+            }
+
+            Console.WriteLine("\nPress any key to exit...");
+            Console.ReadKey(true);
+        }
+
+        /// <summary>
+        /// Get username and password
+        /// </summary>
+        /// <returns></returns>
+        static string[] GetCredencials()
+        {
+            Console.Write("Enter your username: ");
+            string username = Console.ReadLine();
+
+            Console.Write("Enter your password: ");
+            string password = Console.ReadLine();
+
+            return new string[] { username.Replace(" ", String.Empty), password.Replace(" ", String.Empty) };
+        }
+
+        /// <summary>
+        /// Reading Users.txt and saving it's whole content to a dictionary
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        static Dictionary<string, string> GetCredentials(string path)
+        {
+            using (var stream = new StreamReader(path))
+            {
+                string[] fileContent = stream.ReadToEnd().Split(Environment.NewLine);
+
+                Dictionary<string, string> credentials = new Dictionary<string, string>();
+
+                foreach (var item in fileContent)
+                {
+                    if (!String.IsNullOrEmpty(item) || !String.IsNullOrWhiteSpace(item))
```

Comment:
No need to do `IsNullOrEmpty` if you're already doing `IsNullOrWhiteSpace`.  An empty string counts as `IsNullOrWhiteSpace`.